src/providers/ogr/qgsogrfeatureiterator.cpp

```diff
@@ -305,7 +305,7 @@ bool QgsOgrFeatureIterator::fetchFeature( QgsFeature &feature )
     OGRLayerH nextFeatureBelongingLayer;
     while ( fet.reset( GDALDatasetGetNextFeature( mConn->ds, &nextFeatureBelongingLayer, nullptr, nullptr, nullptr ) ), fet )
     {
-      if ( checkFeature( fet, feature ) && QString::compare( QString::fromLatin1( OGR_L_GetName( nextFeatureBelongingLayer ) ), mSource->mLayerName ) == 0 )
+      if ( nextFeatureBelongingLayer == mSource->mLayerName && checkFeature( fet, feature ) )
```

Comment:
huh ! does this actually compile ??? You're comparing a OGRLayer* pointer to a QString !

Reply:
My bad - not at my workstation, so shortcutting. That'll teach me